packages/sdk/src/interaction/interaction.ts

```diff
@@ -104,79 +85,88 @@ const IX = `{
   }
 }`
 
-const KEYS = new Set(Object.keys(JSON.parse(IX)))
+const KEYS = new Set(Object.keys(JSON.parse(IX) as IIx))
 
-export const interaction = () => JSON.parse(IX)
+export const initInteraction = (): IIx => JSON.parse(IX)
 
 const CHARS = "abcdefghijklmnopqrstuvwxyz0123456789".split("")
 const randChar = () => CHARS[~~(Math.random() * CHARS.length)]
 export const uuid = () => Array.from({length: 10}, randChar).join("")
-export const isNumber = d => typeof d === "number"
-export const isArray = d => Array.isArray(d)
-export const isObj = d => d !== null && typeof d === "object"
-export const isNull = d => d == null
-export const isFn = d => typeof d === "function"
+export const isNumber = (d: any): d is number => typeof d === "number"
+export const isArray = (d: any): d is any[] => Array.isArray(d)
+export const isObj = (d: any): d is Record<string, any> => d !== null && typeof d === "object"
+export const isNull = (d: any): d is null => d == null
+export const isFn = (d: any): d is Function => typeof d === "function"
 
-export const isInteraction = ix => {
+export const isInteraction = (ix: IIx) => {
   if (!isObj(ix) || isNull(ix) || isNumber(ix)) return false
   for (let key of KEYS) if (!ix.hasOwnProperty(key)) return false
   return true
 }
 
-export const Ok = ix => {
+export const Ok = (ix: IIx) => {
   ix.status = OK
   return ix
 }
 
-export const Bad = (ix, reason) => {
+export const Bad = (ix: IIx, reason: string) => {
   ix.status = BAD
   ix.reason = reason
   return ix
 }
 
-const makeIx = wat => ix => {
+const makeIx = (wat: string) => (ix: IIx) => {
   ix.tag = wat
   return Ok(ix)
 }
 
-const prepAccountKeyId = acct => {
+const prepAccountKeyId = (acct: Partial<IAcct> | IAcctFn): Partial<IAcct> | IAcctFn => {
   if (acct.keyId == null) return acct
 
-  invariant(!isNaN(parseInt(acct.keyId)), "account.keyId must be an integer")
+  invariant(!isNaN(parseInt(acct.keyId.toString())), "account.keyId must be an integer")
+
   return {
     ...acct,
-    keyId: parseInt(acct.keyId),
-  }
+    keyId: parseInt(acct.keyId.toString()),
+  } as IAcct | IAcctFn
+}
+
+interface IPrepAccountOpts {
+  role?: string | null
```

Comment:
Should this be `typeof AUTHORIZER | typeof PAYER | typeof PROPOSER` instead of asserting this later?